cmd/ftpsource/configstore.go

```diff
@@ -0,0 +1,168 @@
+// Simple abstraction for storing state on a k8s ConfigMap. Very very simple
```

Comment:
FYI, I added different version of configstore (kvstore) to knative.dev/pkg:
https://github.com/knative/pkg/tree/master/kvstore

Maybe use that instead and get rid of this customized version (since I upstreamed it).

Reply:
when the pod starts with side car proxy , initial calls to the api server fail until side car is ready. So using this [func](https://github.com/knative/pkg/blob/ca02ef752ac658f5a315b619e56ec2f87fafdeff/kvstore/kvstore_cm.go#L66) will always return an initial error. How to handle this scenario with kvstore, for now we have the 30s wait and retry logic.

Once the cm is created I switched to informed watcher to read the cm state instead of hitting the api server,  after changing to kv store we can still use this right ?